numpy/core/tests/test_defchararray.py

```diff
@@ -629,6 +629,22 @@ def test_rmod(self):
                 self.fail("chararray __rmod__ should fail with " \
                           "non-string objects")
 
+    def test_slice(self):
+        """Regression test for https://github.com/numpy/numpy/issues/5982"""
+
+        arr = np.array([['abc ', 'def '], ['geh ', 'ijk ']]).view(np.chararray)
+        sl1 = arr[:]
+        assert_array_equal(sl1, arr)
+        assert sl1.base is arr
+        assert sl1.base.base is arr.base
+
+        sl2 = arr[:, :]
+        assert_array_equal(sl2, arr)
+        assert sl2.base is arr
+        assert sl2.base.base is arr.base
+
+        assert arr[0, 0] == b'abc'
```

Comment:
This test fails on python3, because the array contains unicode; add `, dtype='S'` in the definition.